lit_parrot/model.py

```diff
@@ -110,27 +170,30 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         q, k, v = qkv.split(head_size, dim=-1)  # (B, nh, T, hs)
 
         n_elem = int(self.rotary_percentage * head_size)
-        if self.rope_cache is None:
-            self.rope_cache = build_rope_cache(self.block_size, n_elem, x.dtype, x.device)
-        cos, sin = self.rope_cache
-        cos, sin = cos[:T], sin[:T]
 
+        cos, sin = rope
         q_roped = apply_rope(q[..., :n_elem], cos, sin)
         k_roped = apply_rope(k[..., :n_elem], cos, sin)
         q = torch.cat((q_roped, q[..., n_elem:]), dim=-1)
         k = torch.cat((k_roped, k[..., n_elem:]), dim=-1)
 
+        if cache_kv is not None:
+            cache_k, cache_v = cache_kv
+            cache_k = cache_k.index_copy(2, input_pos, k)
+            cache_v = cache_v.index_copy(2, input_pos, v)
+            cache_kv = cache_k, cache_v
+            k = cache_k[:]
+            v = cache_v[:]
```

Comment:
I don't think we need to create a hard copy since `k` and `v` are used for `scaled_dot_product_attention` which shouldn't edits the inputs. I made the adjustment, we can revert if needed.

@Liyang90 do you think this is needed?